applications/RANSApplication/custom_strategies/rans_fractional_step_strategy.h

```diff
@@ -0,0 +1,207 @@
+ * It adds pressure gradient correction to obtain steady state solutions from
+ * Fractional step. This is implemented based on work done by
+ *
+ * A. R. Firoozjaee and M. H. Afshar, Steady-state solution of incompressible navier–stokes
+ * equations using discrete least-squares meshless method, International Journal for Numerical
+ * Methods in Fluids, 67 (2011), pp. 369–382.
+ *
+ * @tparam TSparseSpace Sparse space template type
+ * @tparam TDenseSpace Dense space template type
+ * @tparam TLinearSolver Linear solver template type
+ * @see FractionalStepStrategy
+ */
+template <class TSparseSpace, class TDenseSpace, class TLinearSolver>
+class RansFractionalStepStrategy
+    : public FractionalStepStrategy<TSparseSpace, TDenseSpace, TLinearSolver>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Counted pointer of RansFractionalStepStrategy
+    KRATOS_CLASS_POINTER_DEFINITION(RansFractionalStepStrategy);
+
+    typedef FractionalStepStrategy<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+
+    typedef SolverSettings<TSparseSpace, TDenseSpace, TLinearSolver> SolverSettingsType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    RansFractionalStepStrategy(
+        ModelPart& rModelPart,
+        SolverSettingsType& rSolverConfig,
+        bool PredictorCorrector,
+        bool CalculateReactionsFlag)
+    : BaseType(rModelPart, rSolverConfig, PredictorCorrector, CalculateReactionsFlag)
+    {
+        KRATOS_INFO(this->Info()) << "Created fractional step strategy." << std::endl;
+    }
+
+    RansFractionalStepStrategy(
+        ModelPart& rModelPart,
+        SolverSettingsType& rSolverConfig,
+        bool PredictorCorrector,
+        bool CalculateReactionsFlag,
+        const Kratos::Variable<int>& PeriodicVar)
+    : BaseType(rModelPart, rSolverConfig, PredictorCorrector, CalculateReactionsFlag, PeriodicVar)
+    {
+        KRATOS_INFO(this->Info())
+            << "Created periodic fractional step strategy." << std::endl;
+    }
+
+    /// Destructor.
+    ~RansFractionalStepStrategy() override = default;
+
+    /// Assignment operator.
+    RansFractionalStepStrategy& operator=(RansFractionalStepStrategy const& rOther) = delete;
+
+    /// Copy constructor.
+    RansFractionalStepStrategy(RansFractionalStepStrategy const& rOther) = delete;
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        std::stringstream buffer;
+        buffer << "RansFractionalStepStrategy";
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+
+protected:
+    ///@name Protected Operations
+    ///@{
+
+    std::tuple<bool, double> SolveStep() override
```

Comment:
(maybe we can arrange the implementation in the base class so it fits the needs of both the standard and the RANS formulations).

Reply:
Actually I am doing a simple change, (2 lines). From the following line I calculate `eta`, which is `1.0` for normal FS.
https://github.com/KratosMultiphysics/Kratos/blob/82a21c3d0f63b1fc492631b83410bde3039d78c5/applications/RANSApplication/custom_strategies/rans_fractional_step_strategy.h#L168-L170

Then I apply `eta` in the next line given below.
https://github.com/KratosMultiphysics/Kratos/blob/82a21c3d0f63b1fc492631b83410bde3039d78c5/applications/RANSApplication/custom_strategies/rans_fractional_step_strategy.h#L174

So for standard FS, it would be `1.0 * old_press`, additional multiplication. If it is not a problem, I can integrate this simple change in `FluidDynamicsApplication`.

Reply:
I'd do so. At the end it's a sort of relaxation that we could use outside the RANS framework right?

BTW, I suggest creating a `PRESSURE_RELAXATION` or `FS_PRESSURE_COEFFICIENT` variable in the `FluidDynamicsApplication` to do this. I'm afraid this can be easily misunderstood with the C_p.

(please open an independent PR to do this change)

Reply:
done in #7438